cmd/soroban-cli/src/commands/config/alias.rs

```diff
@@ -0,0 +1,107 @@
+use std::{
+    collections::HashMap,
+    fs::{self, create_dir_all, OpenOptions},
+    io::Write,
+    path::PathBuf,
+};
+
+use serde::{Deserialize, Serialize};
+use stellar_strkey::DecodeError;
+
+use crate::commands::config;
+
+use super::Args;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct Data {
+    ids: HashMap<String, String>,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error("cannot access config dir for alias file")]
+    CannotAccessConfigDir,
+    #[error("cannot parse contract ID {0}: {1}")]
+    CannotParseContractId(String, DecodeError),
+    #[error(transparent)]
+    Config(#[from] config::Error),
+}
+
+impl Args {
+    pub fn load(&self, alias: &str) -> Result<Option<Data>, Error> {
+        let path = self.alias_path(alias)?;
+
+        if !path.exists() {
+            return Ok(None);
+        }
+
+        let content = fs::read_to_string(path)?;
+        let data: Data = serde_json::from_str(&content).unwrap_or_default();
+
+        Ok(Some(data))
+    }
+
+    fn alias_path(&self, alias: &str) -> Result<PathBuf, Error> {
+        let file_name = format!("{alias}.json");
+        let config_dir = self.config_dir()?;
+        Ok(config_dir.join("contract-ids").join(file_name))
+    }
+
+    pub fn save_contract_id(&self, contract_id: &str, alias: Option<&String>) -> Result<(), Error> {
+        let Some(alias) = alias else {
+            return Ok(());
+        };
```

Comment:
I don't feel strongly about this, but the guard probably belongs elsewhere, if the alias is an option and empty, calling code should probably just not call this code.